### PR TITLE
feat: chunked content reading with dynamic overlap, improve performance

### PR DESCRIPTION
**Pull request by andrey18106**
This PR aims to improve performance and memory usage using following approaches:

1. Streamed reading by chunks
2. Use single labels regex applied to chunks step-by-step with possibility to early exit from analysis
3. Adjust config for PDF parser (disable image retention - not needed in our case, set decode memory limit)

### CODE REVIEW

**Comment by marcelklehr** [comment]
This looks good! The tests still need adjusting, I think.

**Review by oleksandr-nc — COMMENTED** [review]
PR looks very nice.

having also tests for the edge cases will be very nice

File: lib/Providers/ContentProviders/MicrosoftContentProvider.php [hunk 1]
```diff
@@ -40,60 +39,77 @@ public function getSupportedMimeTypes(): array {
 
 	/**
 	 * @param \OCP\Files\File $file
-	 * @return string
+	 * @return \Generator<string>
```
[the remark this answers is not in the document]
Comment (andrey18106) [reply lib/Providers/ContentProviders/MicrosoftContentProvider.php:42]: Marcel is correct, empty return kept for readability as an early exit.

File: lib/Providers/ContentProviders/MicrosoftContentProvider.php [hunk 2]
```diff
@@ -40,60 +39,77 @@ public function getSupportedMimeTypes(): array {
...
-				$results = ['headers' => [],'footers' => []];
-
-				foreach ($tree as $child) {
-					if ($child['attributes']['Type'] === 'http://schemas.openxmlformats.org/officeDocument/2006/relationships/footer') {
-						$results['footers'][] = $child['attributes']['Target'];
-					}
-					if ($child['attributes']['Type'] === 'http://schemas.openxmlformats.org/officeDocument/2006/relationships/header') {
-						$results['headers'][] = $child['attributes']['Target'];
+		if ($xml === false) {
+			// Fallback to empty rels if the file is missing
```
[the remark this answers is not in the document]
Comment (andrey18106) [reply lib/Providers/ContentProviders/MicrosoftContentProvider.php:66]: I don't think we should add any extra logging here like there were no logging before.

File: lib/Service/ClassificationService.php [hunk 3]
```diff
@@ -45,20 +49,102 @@ public function getClassificationLabelForFile(File $file) : ?IClassificationLabe
...
+		usort($foundLabels, function (IClassificationLabel $label1, IClassificationLabel $label2) {
 			return $label1->getIndex() <=> $label2->getIndex();
 		});
 
-		return $labels[0];
+		return $foundLabels[0];
+	}
+
+	/**
+	 * @param IClassificationLabel[] $labels
```
[the remark this answers is not in the document]
Comment (andrey18106) [reply lib/Service/ClassificationService.php:69]: Not sure what exactly, it is only PHPDoc comment for $labels array items object type

**Comment by andrey18106** [comment]
PHP cs fixer doesn't support yet 8.4 without force flag yet. Psalm is expected to fail on 8.4, Psalm v6 update is required (in separate PR).
